### PR TITLE
feat(framework): add cleaner and presetter for context

### PR DESCRIPTION
**Pull request by liubog2008**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR fixes** *(optional, close the issue(s) when PR gets merged)*:

Fixes #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by liubog2008** [comment]
@ddysher

**Comment by liubog2008** [comment]
#26

File: cleaner/cleaner.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package cleaner
+
+import "github.com/caicloud/aloe/utils/jsonutil"
+
+// Cleaner defines custom cleaners
```
Comment (ddysher) [inline cleaner/cleaner.go]: Can you give a short definition of what is cleaner here?

File: examples/crud/testdata/_context.yaml [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
-      }
-  response:
-    statusCode: 201
-  definitions:
-  - name: "testProduct"
-  - name: "testProductId"
-    selector:
-    - "id"
\ No newline at end of file
+validatedFlow:
```
[the remark this answers is not in the document]
Comment (liubog2008) [reply examples/crud/testdata/_context.yaml]: Because the roundtrip in the flow has validators = =
Is there any suggestion?

**Comment by liubog2008** [comment]
@ddysher PTAL

File: examples/crud/testdata/_context.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ddysher) [reply examples/crud/testdata/_context.yaml]: nop, just curious..

**Pull request merged**